pytorch3d/renderer/mesh/rasterize_meshes.py

```diff
@@ -98,12 +98,11 @@ def rasterize_meshes(
             # TODO better heuristics for bin size.
             if image_size <= 64:
                 bin_size = 8
-            elif image_size <= 256:
-                bin_size = 16
-            elif image_size <= 512:
-                bin_size = 32
-            elif image_size <= 1024:
-                bin_size = 64
+            else:
+                bin_size = int(2 ** max(np.ceil(np.log2(image_size)) - 4, 4))
+
+    if bin_size != 0:
+        assert 1 + (image_size - 1) // bin_size < 22
```

Comment:
lets change this to raise a `ValueError` with a clear error message as you mentioned in your issue!